omod/src/main/java/org/openmrs/module/owa/web/controller/AddAppController.java

```diff
@@ -95,8 +95,8 @@ public String upload(@RequestParam("file") MultipartFile file, HttpServletReques
                                     }
                                 }
                                 catch(Exception e) {
-                                    message = messageSourceService.getMessage("owa.not_a_zip");
```

Comment:
Why did you replace the above message? It is very different from the one returned by e.getMessage()

Reply:
`e.getMessage()` returns more explicit error message for the error at runtime, where as using `messageSourceService.getMessage()` would mean explicitly defining all possible runtime errors (for all OS environments I suppose) in the messages.properties file and then getting the one that applies for the particular error. This is the approach that impressed me to solve the issue. I'm however open to better alternatives. Thanks.